File: Remove_Control_Flag.js
// Before refactoring
function checkPeople(people) {
  var found = false;
  for (var i = 0; i < people.length; i++) {
    if (!found) {
      if (i == "Mark") {
        found = true;
        sendResult(people[i]);
      }  
      if (i == "John") {
        found = true;
        sendResult(people[i]);
      }  
    }    
  }      
}        
        
// After refactoring
function checkPeople(people) {
  for (var i = 0; i < people.length; i++) {
    if (i == "Mark") {
      sendResult(people[i]);
      break;
    }  
    if (i == "John") {
      sendResult(people[i]);
      break;
    }  
  }    
}        

File: Remove_Control_Flag.py
# Before refactoring
def checkPeople(people):
  found = False;
  for i in people:
    if not found:
      if i == "Mark":
        found = true;
        sendResult(i);
      if i == "John":
        found = true;
        sendResult(i);
        
# After refactoring
def checkPeople(people):
  for i in people:
    if i == "Mark":
      sendResult(i);
      break;
    if i == "John":
      sendResult(i);
      break
